src/agentic/actor_agents.py

```diff
@@ -681,6 +681,60 @@ def get_name(f):
     def handle_request(self, method: str, data: dict):
         return f"Actor {self.name} processed {method} request with data: {data}"
 
+    async def webhook(self, run_id: str, callback_name: str, args: dict) -> Any:
+        """Handle webhook callbacks by executing the specified tool function
+        
+        Args:
+            run_id: ID of the agent run this webhook is for
+            callback_name: Name of the tool function to call
+            args: Arguments to pass to the tool function
+        """
+        # Get the run context from the database
+        db_manager = DatabaseManager()
+        run = db_manager.get_run(run_id)
+        if not run:
+            raise ValueError(f"No run found with ID {run_id}")
+
+        # Recreate run context
+        self.run_context = RunContext(
```

Comment:
this is fine. Note that RunContext can store variables that change while the agent is running. So eventually we should probably persist RunContext and be able to reload it. But we can worry about that later.